File: Programmers/Level2/모음사전.py
# 모음 사전
# 사간 : 15분

import sys
from itertools import product

def solution(word):
    ans = 0
    arr = ["A", "E", "I", "O", "U"]
    wordArr = list()
    for i in range(1, 6):
        for wordTmp in product(arr, repeat = i):
            wordArr.append("".join(wordTmp))
    wordArr.sort()

    for i in range(len(wordArr)):
        if wordArr[i] == word:
            ans = i + 1
            break
    return ans

print(solution("AAAAE"))

File: Programmers/Level2/모음사전.swift
import Foundation

func solution(_ word:String) -> Int {
    let vowel = ["A", "E", "I", "O", "U"]
    var wordArray = [String]()
    var k = 1
    var check = false
    
    func dfs(_ tmpWord: String) {
        if tmpWord != "" {
            wordArray.append(tmpWord)
        }
        if tmpWord.count < 5 {
            for i in 0..<5 {
                if tmpWord + vowel[i] == word {
                    check = true
                }
                if !check {
                    k += 1
                    dfs(tmpWord + vowel[i])
                }
            }
        }
    }
    
    dfs("")
    return k
}

print(solution("AAAAE"))
